release.mk

```diff
@@ -157,16 +171,13 @@ update-changelog:
 	$(SED) 's#head#$(VERSION)#g' CHANGELOG.asciidoc
 
 # Common changelog file steps
-# TODO: changelogs/$(VERSION).asciidoc requires further manipulation to remove empty entries.
 .PHONY: common-changelog
+export CHANGELOG_MINOR_TMPL
 common-changelog: VERSION=$${VERSION}
 common-changelog:
 	@echo ">> common-changelog"
-	mv changelogs/head.asciidoc changelogs/$(VERSION).asciidoc
-	$(SED) 's#head#$(VERSION)#gI' changelogs/$(VERSION).asciidoc
-	$(SED) -E -e 's#(\...)main#\1$(VERSION)#g' changelogs/$(VERSION).asciidoc
-	awk "NR==5{print \"\n* <<release-notes-$(VERSION).0>>\n\n[float]\n[[release-notes-$(VERSION).0]]\n=== APM version $(VERSION).0\"}1" changelogs/$(VERSION).asciidoc > changelogs/$(VERSION).asciidoc.new
-	mv changelogs/$(VERSION).asciidoc.new changelogs/$(VERSION).asciidoc
+	echo "$$CHANGELOG_MINOR_TMPL" > changelogs/$(VERSION).asciidoc
+	tail -n +6 changelogs/head.asciidoc >> changelogs/$(VERSION).asciidoc
```

Comment:
as long as `head.asciidoc` uses the current format, we can use this approach, so less craziness commands to manipulate the file but using templates